include/seqan3/alignment/pairwise/align_pairwise.hpp

```diff
@@ -38,10 +38,92 @@
+ * \param[in] seq             A tuple with two sequences or an input range over such tuples.
+ * \param[in] config          A configuration object containing \ref alignment_configuration
+ *                            "alignment configurations".
+ * \return Depending on the configuration, the return type of align_pairwise is either `void` if the alignment was
+ * configured to call a delegate on each hit or a seqan3::alignment_range over the alignment results.
+ *
+ * \details
+ *
+ * This function computes the pairwise alignment of a sequence pair or a range of sequence pairs.
+ * The type of the elements stored in the tuples must model std::ranges::ForwardRange.
+ * The value category of these elements is either an lvalue or if the passed tuple or range is an lvalue, xvalues are
+ * also allowed as tuple elements.
+ *
+ * For each alignment one or more alignment results can be computed depending on the configuration.
+ * Before computing the alignment the respective alignment algorithm is configured depending on the given alignment
+ * configuration. If the configuration enables a two-way execution (the default case), a seqan3::alignment_range is
+ * returned. The alignment is first computed when calling begin on the returned range or the respective iterator is
+ * incremented.
+ *
+ * ### Example
+ *
+ * The following snippets demonstrate the single element and the range based interface.
+ *
+ * \include test/snippet/alignment/pairwise/align_pairwise.cpp
+ *
+ * ### Exception
+ *
+ * Might throw std::bad_alloc if it fails to allocate the alignment matrix or seqan3::invalid_alignment_configuration
+ * if the configuration is invalid.
+ *
+ * ### Complexity
+ *
+ * The complexity depends on the configured algorithm. For the \ref seqan3::align_cfg::edit "edit distance"
+ * the following worst cases over two input sequences of size \f$ N \f$ can be assumed:
+ *
+ * | Component        | Runtime           | Space            |
+ * |------------------|-------------------|------------------|
+ * | Score only       | \f$ O(N*N/w) \f$  | \f$ O(w) \f$     |
+ * | Back coordinate  | \f$ O(N*N/w) \f$  | \f$ O(w) \f$     |
+ * | Begin coordinate | \f$ O(N*N/w) \f$  | \f$ O(N*N/w) \f$ |
+ * | Alignment        | \f$ O(N*N/w) \f$  | \f$ O(N*N/w) \f$ |
+ *
+ * \f$ w \f$ is the size of a machine word.
+ *
+ * For all other algorithms that compute the standard dynamic programming algorithm the following worst cases hold:
+ *
+ * | Component        | Runtime         | Space          |
+ * |------------------|-----------------|----------------|
+ * | Score only       | \f$ O(N^2) \f$  | \f$ O(N) \f$   |
+ * | Back coordinate  | \f$ O(N^2) \f$  | \f$ O(N) \f$   |
+ * | Begin coordinate | \f$ O(N^2) \f$  | \f$ O(N^2) \f$ |
+ * | Alignment        | \f$ O(N^2) \f$  | \f$ O(N^2) \f$ |
+ *
+ * ### Thread safety
+ *
+ * As long as no shared state is passed to the algorithm via the configuration object, the algorithm is thread safe and
```

Comment:
How can you "pass a shared state to the algorithm via the configuration object"? When would I want to do that and why?

I would be maybe write:

This function is *re-entrant*, i.e. it is always safe to call in parallel **with different inputs**. It is *thread-safe*, i.e. it is safe to call in parallel **with the same input** under the condition that 
1. The input sequences do not change when being iterated over.
2. The configuration ... [does not contain config_element_xy?]